### ISSUE TITLE
nodeos will not stay in sync on 2.0.0-rc2

### ISSUE THREAD

**Issue by matthewdarwin** [EOSIO/eos#8228]
When running nodeos 2.0.0-rc2 on EOS Mainnet on a public p2p endpoint with around 30 incoming connections, nodeos will sometimes not stay in sync.  It sometimes falls behind by 1 or 2 minutes.  After a (a few hours) it will catch up again.  It is not predictable what is going on as it is accepting incoming connections from the general public.

Server is not too fast: Intel(R) Xeon(R) CPU E5-2690 0 @ 2.90GHz
Server has lots of RAM and is using the following configuration
```
wasm-runtime = eos-vm-jit
eos-vm-oc-enable = true
database-map-mode = locked
```
This problem was not experienced on 1.8.x.  (Although of course EOS Mainnet transaction volume does change).  Upgrade to 2.0.0 occurred on Nov 14.

The server is connected to other peers (locally on the same network) that are in sync.

Logs are available if required.

**Comment by sampleblood** [comment]
is this issue solved?

**Comment by matthewdarwin** [comment]
Work in progress.  See https://github.com/EOSIO/eos/pull/8229

**Comment by matthewdarwin** [comment]
The PR referenced above definitely helps.

**Comment by heifner** [comment]
Resolved by #8229
